x-pack/plugins/data_visualizer/public/application/data_comparison/data_comparison_overview_table.tsx

```diff
@@ -0,0 +1,290 @@
+import { COMPARISON_LABEL, DATA_COMPARISON_TYPE_LABEL, REFERENCE_LABEL } from './constants';
+import { useCurrentEuiTheme } from '../common/hooks/use_current_eui_theme';
+import { DataComparisonField, Feature, FETCH_STATUS } from './types';
+import { formatSignificanceLevel } from './data_comparison_utils';
+import { SingleDistributionChart } from './charts/single_distribution_chart';
+import { OverlapDistributionComparison } from './charts/overlap_distribution_chart';
+import { DataComparisonDistributionChart } from './charts/data_comparison_distribution_chart';
+
+const dataComparisonYesLabel = i18n.translate(
+  'xpack.dataVisualizer.dataComparison.fieldTypeYesLabel',
+  {
+    defaultMessage: 'Yes',
+  }
+);
+const dataComparisonNoLabel = i18n.translate(
+  'xpack.dataVisualizer.dataComparison.driftDetectedNoLabel',
+  {
+    defaultMessage: 'No',
+  }
+);
+
+export const DataComparisonOverviewTable = ({
+  data,
+  onTableChange,
+  pagination,
+  sorting,
+  status,
+}: {
+  data: Feature[];
+  status: FETCH_STATUS;
+} & UseTableState<Feature>) => {
+  const euiTheme = useCurrentEuiTheme();
+  const colors = {
+    referenceColor: euiTheme.euiColorVis2,
+    productionColor: euiTheme.euiColorVis1,
+  };
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, ReactNode>>(
+    {}
+  );
+
+  const referenceDistributionLabel = i18n.translate(
+    'xpack.dataVisualizer.dataComparison.dataComparisonDistributionLabel',
+    {
+      defaultMessage: '{label} distribution',
+      values: { label: REFERENCE_LABEL },
+    }
+  );
+  const comparisonDistributionLabel = i18n.translate(
+    'xpack.dataVisualizer.dataComparison.dataComparisonDistributionLabel',
+    {
+      defaultMessage: '{label} distribution',
+      values: { label: COMPARISON_LABEL },
+    }
+  );
+
+  const columns: Array<EuiBasicTableColumn<Feature>> = [
+    {
+      align: 'left',
+      width: '40px',
+      isExpander: true,
+      name: (
+        <EuiScreenReaderOnly>
+          <span>{EXPAND_ROW}</span>
+        </EuiScreenReaderOnly>
+      ),
+      render: (item: Feature) => {
+        const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+
+        return (
+          <EuiButtonIcon
+            onClick={() => toggleDetails(item)}
+            aria-label={itemIdToExpandedRowMapValues[item.featureName] ? COLLAPSE_ROW : EXPAND_ROW}
+            iconType={itemIdToExpandedRowMapValues[item.featureName] ? 'arrowDown' : 'arrowRight'}
+          />
+        );
+      },
+    },
+
+    {
+      field: 'featureName',
+      name: i18n.translate('xpack.dataVisualizer.dataComparison.fieldNameLabel', {
+        defaultMessage: 'Field name',
+      }),
+      'data-test-subj': 'mlDataComparisonOverviewTableFeatureName',
+      sortable: true,
+      textOnly: true,
+    },
+    {
+      field: 'fieldType',
+      name: i18n.translate('xpack.dataVisualizer.dataComparison.fieldTypeLabel', {
+        defaultMessage: 'Field type',
```

Comment:
In the Data Visualizer we use `Type` here, and again we should be consistent on naming. Is this showing the same values as in the Data Visualizer? If so, any thoughts on using the icons as we use in that view and in Discover?

Reply:
Did you look at using the same icons as in the Data Visualizer / Discover for indicating the field type, rather than using the terms 'Categorical' and 'Numeric'>

Reply:
Updated here [`0622573` (#161365)](https://github.com/elastic/kibana/pull/161365/commits/0622573550793287dc8bc1d74788c9ab112268f1) 
<img width="1299" alt="Screen Shot 2023-07-26 at 11 27 58" src="https://github.com/elastic/kibana/assets/43350163/173cb52a-72f9-4d03-ada2-5ccc28e3a712">